### PR TITLE
feat: create watchable resources using the client

### PR DESCRIPTION
**Pull request by kamiiiel**
**Issue**

https://gravitee.atlassian.net/browse/APIM-2969

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `3.1.0-create-resorce-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/kubernetes/gravitee-kubernetes/3.1.0-create-resorce-SNAPSHOT/gravitee-kubernetes-3.1.0-create-resorce-SNAPSHOT.zip)

### CODE REVIEW

File: gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/KubernetesSecretV1Test.java [hunk 1]
```diff
@@ -463,6 +464,20 @@ public void shouldRetryWatchOnConnectionError() throws InterruptedException {
...
+        io.gravitee.kubernetes.client.model.v1.Secret secret = new io.gravitee.kubernetes.client.model.v1.Secret();
+        io.gravitee.kubernetes.client.model.v1.ObjectMeta metadata = new io.gravitee.kubernetes.client.model.v1.ObjectMeta();
+        metadata.setName("test");
+        metadata.setNamespace("test");
+        secret.setMetadata(metadata);
+
+        server.expect().post().withPath("/api/v1/namespaces/test/secrets").andReturn(201, secret).once();
+
+        TestObserver<Watchable> obs = kubernetesClient.create(secret).test();
+        obs.await();
```
Comment (benoitgravitee) [inline gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/KubernetesSecretV1Test.java:478]: A timeout and an assertion would be preferable

File: gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/KubernetesConfigMapV1Test.java [hunk 2]
```diff
@@ -278,6 +279,20 @@ public void shouldRetryWatchOnConnectionError() throws InterruptedException {
...
+        io.gravitee.kubernetes.client.model.v1.ConfigMap configMap = new io.gravitee.kubernetes.client.model.v1.ConfigMap();
+        io.gravitee.kubernetes.client.model.v1.ObjectMeta metadata = new io.gravitee.kubernetes.client.model.v1.ObjectMeta();
+        metadata.setName("test");
+        metadata.setNamespace("test");
+        configMap.setMetadata(metadata);
+
+        server.expect().post().withPath("/api/v1/namespaces/test/configmaps").andReturn(201, configMap).once();
+
+        TestObserver<Watchable> obs = kubernetesClient.create(configMap).test();
+        obs.await();
```
Comment (benoitgravitee) [inline gravitee-kubernetes-client/src/test/java/io/gravitee/kubernetes/client/KubernetesConfigMapV1Test.java:293]: A timeout and an assertion would be preferable

File: gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java [hunk 3]
```diff
@@ -75,24 +79,67 @@ public KubernetesClientV1Impl(KubernetesConfig kubeConfig) {
...
+            RequestOptions requestOptions = getHTTPRequestOptions(HttpMethod.POST, uri);
+
+            ObjectMapper objectMapper = new ObjectMapper();
+            byte[] body = objectMapper.writeValueAsBytes(item);
+            return httpClient()
+                .rxRequest(requestOptions)
+                .flatMap(httpClientRequest -> httpClientRequest.rxSend(io.vertx.rxjava3.core.buffer.Buffer.buffer(body)))
+                .toMaybe()
+                .flatMap(response -> {
+                    if (response.statusCode() > 299) {
```
Comment (benoitgravitee) [inline gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java:103]: 3xx are not errors. Are you sure the client does not follow redirect ?

File: gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClient.java [hunk 4]
```diff
@@ -28,6 +28,8 @@
  * @since 3.9.11
  */
 public interface KubernetesClient {
+    Maybe<Watchable> create(Watchable item);
```
Comment (benoitgravitee) [inline gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/KubernetesClient.java:31]: Some javadoc on all method, even it sounds trivial

**Review by benoitgravitee — CHANGES_REQUESTED** [review]

File: gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java (hunk 3 above)
Comment (kamiiiel) [reply gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java:103]: There might be 3 responses. I can change this IF the way you prefer :-)
200 OK
201 Created
202  Accepted

Comment (benoitgravitee) [reply gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java:103]: I think I'd be clearer, sorry :-D

Comment (kamiiiel) [reply gravitee-kubernetes-client/src/main/java/io/gravitee/kubernetes/client/impl/KubernetesClientV1Impl.java]: Hope it is fine now :-)

**Review by benoitgravitee — APPROVED** [review]

**Pull request merged**